src/librustc_trans/trans/datum.rs

```diff
@@ -496,8 +509,10 @@ impl<'tcx> Datum<'tcx, Rvalue> {
 
             ByValue => {
                 lvalue_scratch_datum(
-                    bcx, self.ty, name, scope, self,
+                    bcx, self.ty, name, InitAlloca::Dropped, scope, self,
                     |this, bcx, llval| {
+                        debug!("populate call for Datum::to_lvalue_datum_in_scope \
+                                self.ty={:?}", this.ty);
                         call_lifetime_start(bcx, llval);
```

Comment:
Okay; I will take care not to introduce multiple lifetime starts on a given path. (This will complicate the PR a bit but so be it)